### PR TITLE
fix: deleting a custom provider can't leave orphaned data

### PR DESCRIPTION
**Pull request by brunobuddy**
### 💭 Why
A custom provider is two rows (config + stored key) linked only by an application-managed string. Deleting one side could orphan the other, and a failure mid-create could strand a half-created provider. This already bit us once: #1603 needed a cleanup migration after deleted custom providers kept receiving routed requests.

### ✨ What changed
- New migration adds a real FK between the two tables via a stored generated column, with cascade delete.
- The migration removes pre-existing orphan key rows (unresolvable, invisible in the UI); reachable rows are untouched.
- Creating and deleting a custom provider now run both rows in one transaction.
- Two new e2e suites run the real migration chain against Postgres: whole-table snapshots prove nothing changes except orphan removal, plus FK rejection, cascade, and rollback checks.

### 🔧 For operators
Migration runs automatically on boot. No env vars or deploy-order constraints. Rollback (`migration:revert`) drops the FK and column; deleted orphans are not restored (they were unreachable).

### 📝 Notes
Edge cases pinned by tests: empty-id keys (`custom:`), inactive keys, multiple keys per provider with different auth types, agent grants on surviving vs orphaned keys.

---
## Summary by cubic
Deleting a custom provider can no longer leave orphaned keys or routing/grant data. We enforce a real DB link and make create/delete atomic to protect data integrity.

- **Bug Fixes**
  - Added a stored generated column and real FK from `user_providers` to `custom_providers` with ON DELETE CASCADE, plus a partial index.
  - One-time cleanup removes pre-existing orphan companion rows (including empty-id `custom:`); reachable rows are untouched.
  - `CustomProviderService.create/remove` run both rows in one transaction; `ProviderService` now threads an optional `EntityManager` so all `user_providers` and `agent_enabled_providers` writes (including labeled-key paths) join the same tx.
  - New tests: migration unit, ProviderService tx-manager threading, and Postgres e2e suites proving no data loss, FK enforcement, cascade, and rollback.

- **Migration**
  - Runs automatically on boot; no config changes required.
  - Revert drops the FK and helper column; deleted orphans are not restored.

<sup>Written for commit 30539c2312ba3a81a704310912eb70671413b448. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/mnfst/manifest/pull/2238?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: .changeset/global-dashboard-overview.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"manifest": patch
+---
+
+Add global dashboard with Overview, Messages, and Agents navigation for tenant-wide usage analytics.
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply .changeset/global-dashboard-overview.md:5]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: .changeset/agent-provider-chart.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'manifest': patch
+---
+
+The per-agent Overview now breaks usage down by provider (chart + provider filter), matching the global Overview.
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply .changeset/agent-provider-chart.md:5]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: .changeset/agent-tabbed-shell.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"manifest": patch
+---
+
+Restructure agent detail view into a horizontal-tabbed shell (Overview / Routing / Guardrails / Settings) with back-link navigation. Unify sidebar to a single global nav (Overview / Messages / Agents) rendered identically on every authenticated route; remove the agent-scoped MONITORING/MANAGE/RESOURCES sub-nav.
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply .changeset/agent-tabbed-shell.md:5]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: packages/backend/src/playground/playground-agent.service.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          `INSERT INTO "agent_provider_access" ("agent_id","user_provider_id") ` +
+            `SELECT $1, "id" FROM "user_providers" WHERE "user_id" = $2 ON CONFLICT DO NOTHING`,
+          [agent.id, userId],
+        );
+      });
+    } catch (err) {
+      // A concurrent request may have won the unique-index race — reuse its row.
+      // Any other failure is a real error (not a race): re-throw it rather than
+      // masking it as a generic not-found.
+      const raced = await this.findSystemAgent(tenantId);
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply packages/backend/src/playground/playground-agent.service.ts]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: .changeset/agent-messages-filter-redirect.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'manifest': patch
+---
+
+"View more" on a harness's recent messages now opens the global Messages log pre-filtered to that harness.
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply .changeset/agent-messages-filter-redirect.md:5]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: packages/backend/src/notifications/services/notification-rules.service.ts [hunk 6]
```diff
@@ -123,12 +123,18 @@ export class NotificationRulesService {
...
+          at.agent_id = (
+            SELECT id FROM agents
+            WHERE tenant_id = at.tenant_id
+              AND name = :agentName
+              AND deleted_at IS NULL
+            LIMIT 1
+          )
+          OR (
+            at.agent_id IS NULL
+            AND at.agent_name = :agentName
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply packages/backend/src/notifications/services/notification-rules.service.ts:136]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: packages/backend/src/database/migrations/1791400000000-SeedPlaygroundAgents.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+       JOIN "user_providers" up ON up."user_id" = t."name"
+       WHERE a."is_system" = true AND a."deleted_at" IS NULL
+       ON CONFLICT DO NOTHING`,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    // Grants cascade-delete with the agent rows (agent_provider_access FK is
+    // ON DELETE CASCADE), so removing the system agents is enough.
+    await queryRunner.query(`DELETE FROM "agents" WHERE "is_system" = true`);
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply packages/backend/src/database/migrations/1791400000000-SeedPlaygroundAgents.ts:56]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.

File: packages/backend/src/database/migrations/1791400000000-SeedPlaygroundAgents.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    );
+
+    // 2. Free the reserved name: relabel any existing NON-system agent literally
+    //    named 'Playground' (suffix with its id — never delete a user's agent) so
+    //    the reserved per-tenant agent can take the name without colliding on the
+    //    (tenant_id, name) unique index.
+    //    Slug-safe form: `name-<uuid>` (hyphen + UUID chars are all [a-z0-9-]).
+    //    Space/bracket suffixes would break ^[a-zA-Z0-9_-]+$ route validators.
+    await queryRunner.query(
+      `UPDATE "agents" SET "name" = "name" || '-' || "id"
```
[the remark this answers is not in the document]
Comment (brunobuddy) [reply packages/backend/src/database/migrations/1791400000000-SeedPlaygroundAgents.ts:20]: Stale — this file is not part of this PR. The comment was generated while the PR was briefly mis-based on main, which pulled in unrelated next-branch commits. The PR is now correctly based on next (+1,240/−73); only the custom-provider FK migration, transactional create/delete, and their tests remain.
